tst/com/amazon/corretto/crypto/provider/test/RsaCipherTest.java

```diff
@@ -997,6 +995,44 @@ public void noInputDoFinal() throws Exception {
     }
   }
 
+  @Test
+  public void testOaepCustomP() throws Exception {
+    final OAEPParameterSpec customPSource =
+        new OAEPParameterSpec(
+            OAEPParameterSpec.DEFAULT.getDigestAlgorithm(),
+            OAEPParameterSpec.DEFAULT.getMGFAlgorithm(),
+            OAEPParameterSpec.DEFAULT.getMGFParameters(),
+            new PSource.PSpecified(new byte[] {0x00, 0x01, 0x02, 0x03}));
+    final OAEPParameterSpec defaultParamsSpec =
+        new OAEPParameterSpec(
+            OAEPParameterSpec.DEFAULT.getDigestAlgorithm(),
+            OAEPParameterSpec.DEFAULT.getMGFAlgorithm(),
+            OAEPParameterSpec.DEFAULT.getMGFParameters(),
+            PSource.PSpecified.DEFAULT);
+    final byte[] plaintext =
+        getPlaintext((2048 / 8) - getPaddingSize(OAEP_PADDING, customPSource) - 4);
+    final Cipher encrypt = getNativeCipher(OAEP_PADDING);
+    final Cipher decrypt = getNativeCipher(OAEP_PADDING);
+
+    // Encrypt/decrypt with same PSource should succeed
+    encrypt.init(Cipher.ENCRYPT_MODE, PAIR_2048.getPublic(), customPSource);
+    decrypt.init(Cipher.DECRYPT_MODE, PAIR_2048.getPrivate(), customPSource);
+    byte[] ciphertext = encrypt.doFinal(plaintext);
+    byte[] decrypted = decrypt.doFinal(ciphertext);
+    assertArrayEquals(plaintext, decrypted);
+
+    // Encrypt/decrypt with differing PSource should fail
+    encrypt.init(Cipher.ENCRYPT_MODE, PAIR_2048.getPublic(), customPSource);
+    decrypt.init(Cipher.DECRYPT_MODE, PAIR_2048.getPrivate(), defaultParamsSpec);
```

Comment:
nit: Rather than `defaultParamsSpec`, I think having a customParamSpec with the wrong values would be a stronger assertion.